sdk/cognitiveservices/azopenai/models.go

```diff
@@ -237,9 +237,8 @@ type ChatCompletionsOptions struct {
 	// The maximum number of tokens to generate.
 	MaxTokens *int32
 
-	// The model name to provide as part of this completions request. Not applicable to Azure OpenAI, where deployment information
-	// should be included in the Azure resource URI that's connected to.
-	Model *string
+	// DeploymentID specifies the name of the deployment (for Azure OpenAI) or model (for OpenAI) to use for this request.
+	DeploymentID string
```

Comment:
I guess the lack of * is OK here since it really becomes mandatory. But maybe not for images?

Reply:
Presumably it's because the empty string and `nil` are equivalent in this context?

Reply:
Images doesn't have the model/deploymentID parameter, so we're safe there.

It's a required parameter, so you get bad request from OpenAI and deployment not found from Azure OpenAI if it's empty. TBH, we might want to check this on the client side but the service-side errors aren't horrible either.

Reply:
Oh this is the options type that isn't our synthesized options type right?  Does this need a comment that it's required for OpenAI?